src/Splat.SimpleInjector/SimpleInjectorInitializer.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) 2019 .NET Foundation and Contributors. All rights reserved.
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Splat.SimpleInjector
+{
+    /// <summary>
+    /// sad.
+    /// </summary>
+#pragma warning disable CA1063 // Implement IDisposable Correctly
+    public class SimpleInjectorInitializer : IDependencyResolver
+#pragma warning restore CA1063 // Implement IDisposable Correctly
+    {
+        /// <summary>
+        /// Gets dictionary of registered factories.
+        /// </summary>
+        public Dictionary<Type, List<Func<object>>> RegisteredFactories { get; }
+            = new Dictionary<Type, List<Func<object>>>();
+
+        /// <inheritdoc />
+        public object GetService(Type serviceType, string contract = null)
+        {
+            Func<object> fact = RegisteredFactories[serviceType].LastOrDefault();
```

Comment:
Thread safety tweak needed with lock see https://github.com/reactiveui/splat/blob/7af765b6ac360c93dc217d6380cc045c23c2401a/src/Splat.Autofac/AutofacDependencyResolver.cs#L55-L61 for example